extensions/websocket/internal.m

```diff
@@ -0,0 +1,294 @@
+@import Foundation;
+@import Cocoa;
+@import Carbon;
+@import LuaSkin;
+@import WebKit;
+#import <SocketRocket/SRWebSocket.h>
+
+// Websocket userdata struct
+typedef struct _webSocketUserData {
+    int selfRef;
+    void *ws;
+} webSocketUserData;
+
+#define getWsUserData(L, idx) (__bridge HSWebSocketDelegate *)((webSocketUserData *)lua_touserdata(L, idx))->ws;
+static const char *WS_USERDATA_TAG = "hs.websocket";
+
+static int refTable;
+
+@interface HSWebSocketDelegate: NSObject<SRWebSocketDelegate>
+@property int fn;
+@property (strong) SRWebSocket *webSocket;
+@end
+
+@implementation HSWebSocketDelegate
+
+- (instancetype)initWithURL:(NSURL *)URL {
+    if((self = [super init])) {
+        _webSocket = [[SRWebSocket alloc] initWithURL:URL];
+        _webSocket.delegate = self;
+    }
+    return self;
+}
+- (void)webSocket:(SRWebSocket *)webSocket didReceiveMessage:(id)message {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (self.fn == LUA_NOREF) {
+            return;
+        }
+        LuaSkin *skin = [LuaSkin shared];
```

Comment:
All of the `[LuaSkin shared]` should be updated to the new `[LuaSkin sharedWithState:NULL]`.

Reply:
Done